### ISSUE TITLE
[Unstable 0.11.0.8] Game crashes if you control a Black Moloch and attack the submarine

### ISSUE THREAD

**Issue by Rokvach** [Regalis11/Barotrauma#4393]
Repro steps:
1. Start a singleplayer round with Dugong
2. Spawn in a Black Moloch
3. Control it and attack the hull (Will take some attacks before the crash happens)

```
Barotrauma Client crash report (generated on 27.11.2020 1:53:34)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


7C69FEB1E8077B8F7063C8245691EA70


Game version 0.11.0.8 (ReleaseWindows, branch release, revision b33722ad5b)
Graphics mode: 2560x1440 (BorderlessWindowed)
VSync OFF
Language: English
Selected content packages: Vanilla 0.9
Level seed: Sarpedon LineaNovaya Moskva
Loaded submarine: Dugong (5042252D50B7FFEBC2ED1A64DD1F0B23)
Selected screen: Barotrauma.GameScreen
SteamManager initialized


System info:
    Operating system: Microsoft Windows NT 10.0.18363.0 64 bit
    GPU name: NVIDIA GeForce GTX 980 Ti
    Display mode: {Width:2560 Height:1440 Format:Color AspectRatio:1,7777778}
    GPU status: Normal


Exception: Value cannot be null. (Parameter 'key') (System.ArgumentNullException)
Target site: Void ThrowArgumentNullException(System.ExceptionArgument)
Stack trace: 
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   at Barotrauma.Explosion.DamageCharacters(Vector2 worldPosition, Attack attack, Single force, Entity damageSource, Character attacker) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Map\Explosion.cs:line 255
   at Barotrauma.Explosion.Explode(Vector2 worldPosition, Entity damageSource, Character attacker) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Map\Explosion.cs:line 174
   at Barotrauma.StatusEffect.Apply(Single deltaTime, Entity entity, IEnumerable`1 targets, Nullable`1 worldPosition) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\StatusEffects\StatusEffect.cs:line 965
   at Barotrauma.StatusEffect.Apply(ActionType type, Single deltaTime, Entity entity, ISerializableEntity target, Nullable`1 worldPosition) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\StatusEffects\StatusEffect.cs:line 778
   at Barotrauma.Attack.DoDamage(Character attacker, IDamageable target, Vector2 worldPosition, Single deltaTime, Boolean playSound, PhysicsBody sourceBody) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Attack.cs:line 440
   at Barotrauma.Limb.ExecuteAttack(IDamageable damageTarget, Limb targetLimb, AttackResult& attackResult) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Limb.cs:line 1024
   at Barotrauma.Limb.UpdateAttack(Single deltaTime, Vector2 attackSimPos, IDamageable damageTarget, AttackResult& attackResult, Single distance, Limb targetLimb) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Limb.cs:line 941
   at Barotrauma.Character.Control(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 1552
   at Barotrauma.Character.Update(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 2502
   at Barotrauma.AICharacter.Update(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AICharacter.cs:line 81
   at Barotrauma.Character.UpdateAll(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 2318
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 183
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 924
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 656
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 504
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 94
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 398
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 368
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 59


Last debug messages:
[27.11.2020 1:53:06] control Molochblack
[27.11.2020 1:53:00] spawncharacter molochblack cursor
[27.11.2020 1:52:48] Lighting disabled
[27.11.2020 1:52:48] lighting
[27.11.2020 1:52:46] freecamera
[27.11.2020 1:52:43] Steam achievements have been disabled during this play session.
[27.11.2020 1:52:43] Enabled cheat commands.
[27.11.2020 1:52:43] enablecheats
[27.11.2020 1:52:28] Generating an outpost for the start of the level... (Location: Sarpedon Linea, level type: LocationConnection)
[27.11.2020 1:52:19] Transitioning to Sarpedon LineaNovaya Moskva (current location: Novaya Moskva Station, selected location: Sarpedon Linea, leaving sub: Dugong, at start: True, at end: False, transition type: LeaveLocation)
[27.11.2020 1:52:04] Generating an outpost for the start of the level... (Location: Novaya Moskva Station, level type: Outpost)
[27.11.2020 1:52:01] (Msg) SDR RelayNetworkStatus:  avail=OK  config=OK  anyrelay=OK   (OK)
[27.11.2020 1:52:01] (Msg) Ping location: sto2=14+1,sto=15+1/14+1,fra=40+4/33+1,ams=39+3/34+1,waw=45+4/34+1,lux=47+4/37+1,lhr=47+4/38+1,par=51+5/42+1,vie=52+5/45+1,iad=129+12/119+1,sgp=197+19/198+19,gru=235+23
[27.11.2020 1:52:01] (Msg) Ping measurement completed
[27.11.2020 1:51:59] (Debug) Destroying relay 'sof#3 (139.45.193.10:27017)' because initial_ping_timeout
[27.11.2020 1:51:56] (Debug) Destroying relay 'sof#1 (139.45.193.10:27015)' because initial_ping_timeout
[27.11.2020 1:51:54] (Debug) Destroying relay 'sof#5 (139.45.193.10:27019)' because initial_ping_timeout
[27.11.2020 1:51:51] (Debug) Destroying relay 'sof#2 (139.45.193.10:27016)' because initial_ping_timeout
[27.11.2020 1:51:51] WARNING: Cannot compress a texture because the dimensions are not a multiple of 4 (path: Submarines/Downloaded/Khors.sub, size: 700x313)
[27.11.2020 1:51:50] WARNING: Cannot compress a texture because the dimensions are not a multiple of 4 (path: Submarines/Downloaded/Inglia.sub, size: 700x313)
[27.11.2020 1:51:50] WARNING: Cannot compress a texture because the dimensions are not a multiple of 4 (path: Submarines/Downloaded/Indra.sub, size: 700x313)
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'syd' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'tyo1' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'tyo' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'sgp' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'sea' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'lim' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'scl' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'hkg' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'maa' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'lax' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'okc' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'vie' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'waw' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'gru' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'ord' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'jnb' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'sto2' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'sto' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'eat' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'mad' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'bom' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'dxb' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'par' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'lux' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'lhr' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'atl' established.
[27.11.2020 1:51:49] (Verbose) Communications with routing cluster 'iad' established.
[27.11.2020 1:51:48] (Verbose) Communications with routing cluster 'fra' established.
[27.11.2020 1:51:48] (Verbose) Communications with routing cluster 'ams' established.
[27.11.2020 1:51:48] (Msg) SDR RelayNetworkStatus:  avail=Attempting  config=OK  anyrelay=Attempting   (Performing ping measurement)
[27.11.2020 1:51:48] (Msg) Performing ping measurement
[27.11.2020 1:51:48] (Msg) Got SDR network config.  Loaded revision 285 OK
[27.11.2020 1:51:48] (Warning) SDR network config fetch first attempt failed.  HTTP 504.  .  Trying again.
[27.11.2020 1:51:42] Attempting to open ALC device "OpenAL Soft on Kaiuttimet (Realtek High Definition Audio)"
[27.11.2020 1:51:41] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
[27.11.2020 1:51:41] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
```

**Comment by muskaeragon** [comment]
This bug is also present in the normal version.

**Comment by itchyOwl** [comment]
Fixed in Regalis11/Barotrauma-development@a24d391

**Comment by Rokvach** [comment]
Tested, wasn't able to reproduce the issue. Working correctly. Closing.
